src/multi.c

```diff
@@ -259,11 +259,12 @@ void unwatchAllKeys(client *c) {
          * from the list */
         wk = listNodeValue(ln);
         clients = dictFetchValue(wk->db->watched_keys, wk->key);
-        serverAssertWithInfo(c,NULL,clients != NULL);
-        listDelNode(clients,listSearchKey(clients,c));
-        /* Kill the entry at all if this was the only client */
-        if (listLength(clients) == 0)
-            dictDelete(wk->db->watched_keys, wk->key);
+        if (clients) {
```

Comment:
Can you explain in what case `clients` can be `NULL` here?

Reply:
Oh, this PR is too old ; ) after rethinking of it, the case is in `touchWatchedKey()` I already deleted the key in `db->watched_keys` dict, so the value (a list of clients) is NULL.